src/url_aggregator.cpp

```diff
@@ -99,4 +99,8 @@ bool url_aggregator::set_hostname(const std::string_view input) {
   return components;
 }
 
+std::string ada::url_aggregator::to_string() const {
+  return components.to_string();
```

Comment:
Is that what you really want from to_string() in this case ?

(Honest question: I have no opinion.)

Reply:
Well, I think this behaves like a `print()` or `debug_print()`. I don't think we need a to_string.

Reply:
I have no opinion, I was just raising the point for discussion.